### PR TITLE
Adapt account creation infos to exotic paths 

### PR DESCRIPTION
**Pull request by KhalilBellakrid**
- The next creation infos that wallets were returning were not adapted to exotic derivation schemes supported by Live,
- Add tests for weird derivation schemes

### CODE REVIEW

File: core/src/wallet/ethereum/EthereumLikeWallet.cpp [hunk 1]
```diff
@@ -148,6 +148,22 @@ namespace ledger {
...
+            auto accountScheme = scheme.getSchemeTo(DerivationSchemeLevel::ACCOUNT_INDEX);
+            // To handle all exotic paths we should avoid private derivations
+            // So if node or/and address level are hardened, then they are included in account's derivation path
+            auto path = scheme.getPath();
+            auto hardenedDepth = path.getDepth();
+            while (hardenedDepth) {
+                if (path.isHardened(hardenedDepth - 1)) {
+                    break;
+                }
+                hardenedDepth --;
```
Comment (phaazon) [inline core/src/wallet/ethereum/EthereumLikeWallet.cpp]: Can you use either `hardenedDepth--` or `--hardenedDepth` but not `hardenedDepth --`, please? 🙂

File: core/src/utils/DerivationScheme.cpp [hunk 2]
```diff
@@ -89,7 +89,9 @@ namespace ledger {
                                                                        DerivationSchemeLevel::NODE,
                                                                        DerivationSchemeLevel::ADDRESS_INDEX};
                 for (size_t i = 0; i < _scheme.size(); i++ ) {
-                    if (_scheme[i].level == DerivationSchemeLevel::UNDEFINED) {
+                    // We don't set twice a level
+                    auto it = std::find_if(_scheme.begin() + i, _scheme.end(), [&](const DerivationSchemeNode &node){return node.level == schemeLevel[i];});
```
[the remark this answers is not in the document]
Comment (gre) [reply core/src/utils/DerivationScheme.cpp]: closures. i didn't even know it was possible in C++.

[the remark this answers is not in the document]
Comment (phaazon) [reply core/src/utils/DerivationScheme.cpp]: It’s been possible since C++11 IIRC :)

[the remark this answers is not in the document]
Comment (KhalilBellakrid) [reply core/src/utils/DerivationScheme.cpp]: It's fine here taking into account schemes fixed size (5)

**Pull request merged**